include/mapbox/geojson_impl.hpp

```diff
@@ -81,8 +88,12 @@ geometry convert<geometry>(const rapidjson_value &json) {
         return geometry{ convert<point>(json_coords) };
     if (type == "MultiPoint")
         return geometry{ convert<multi_point>(json_coords) };
-    if (type == "LineString")
+    if (type == "LineString") {
+        if (json_coords.GetArray().Size() < 2) {
```

Comment:
Not quite sure if this validity check is necessary, since it is only value converter instead of value validator, otherwise, you will have to validate the following `MultiLineString`, `Polygon` and `MultiPolygon` cases also.

cc: @alexshalamov @pozdnyakov